libvast/src/arrow_table_slice_builder.cpp

```diff
@@ -143,16 +144,19 @@ table_slice arrow_table_slice_builder::finish() {
     caf::get<type_to_arrow_array_t<record_type>>(*combined_array).fields());
   // Reset the builder state.
   num_rows_ = {};
-  return create_table_slice(record_batch, this->builder_,
+  return create_table_slice(record_batch, this->builder_, layout(),
                             table_slice::serialize::yes);
 }
 
 table_slice arrow_table_slice_builder::create(
-  const std::shared_ptr<arrow::RecordBatch>& record_batch,
+  const std::shared_ptr<arrow::RecordBatch>& record_batch, type schema,
   table_slice::serialize serialize, size_t initial_buffer_size) {
+#if VAST_ENABLE_ASSERTIONS
```

Comment:
Can we just do VAST_ASSERT(verify_record_batch(*record_batch)) ?

Reply:
yeah, nice idea. had to change the return type of `verify_record_batch` to `bool`, but that's fine I guess.